### PR TITLE
Support showing autofill surveys based on remote config

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1207935899042228/f 

### Description
Adds support for consuming autofill surveys from remote config, and converts the autofill survey UI to make use of the new `PasswordsScreenPromotionPlugin` framework.

### Steps to test this PR

**Test nothing has broken when no surveys available**
- [x] Fresh install, visit `Passwords` screen
- [x] Verify there is no survey showing. 
- [x] Verify there is no sync promo showing.
- [x] Add a password (manually or through autofill), then return visit `Passwords` screen again; verify sync promo shows

**Test survey shows when it is available**
- [x] Apply patch to fake a remote config survey being available (see patch below)
- [x] Launch the app and visit `Passwords` screen; verify the survey shows
- [x] Verify `m_autofill_management_screen_visit_survey_available` pixel shows in logs

**Test dismissing the survey**
- [x] Dismiss the survey prompt; verify it goes away
- [x] Verify you see the sync promo (if you still have >= 1 saved passwords)

**Test taking the survey**
- [x] Either do a fresh install, or use `Settings` -> `Autofill Dev Settings` -> `Previously Seen Surveys` to reset state
- [x] Visit `Passwords` screen, and tap `Take Survey` button
- [x] Verify it loads in a tab and the survey is shown (don't worry about the survey contents itself; still being refined)
- [x] Return to the `Passwords` screen; verify the survey prompt does **not** show

### UI changes
combined

### Patch for testing

Right now, there is no survey active in remote config, so you can use this patch to see a survey.

```
Index: privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt b/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt
--- a/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt	(revision Staged)
+++ b/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt	(date 1725364069768)
@@ -27,4 +27,5 @@
     TrackingParametersFeatureName("trackingParameters"),
 }
 
-const val PRIVACY_REMOTE_CONFIG_URL = "https://staticcdn.duckduckgo.com/trackerblocking/config/v4/android-config.json"
+// const val PRIVACY_REMOTE_CONFIG_URL = "https://staticcdn.duckduckgo.com/trackerblocking/config/v4/android-config.json"
+const val PRIVACY_REMOTE_CONFIG_URL = "https://jsonblob.com/api/1280153728681107456"

```

### CODE REVIEW

**Comment by CDRussell** [comment]
* **#4981** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4981?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CDRussell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/promotion/passwords/SyncPasswordsPromotion.kt [hunk 1]
```diff
@@ -51,6 +52,7 @@ import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
 
 @ContributesMultibinding(scope = AppScope::class)
+@PriorityKey(200)
```
Comment (joshliebe) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/promotion/passwords/SyncPasswordsPromotion.kt:55]: Why did we go with 200? Is there a `PasswordsScreenPromotionPlugin` with priority 100? (Should it be a constant?)

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/survey/SurveyInPasswordsPromotion.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import dagger.android.support.AndroidSupportInjection
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@ContributesMultibinding(scope = AppScope::class)
+@PriorityKey(100)
```
Comment (joshliebe) [inline autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/survey/SurveyInPasswordsPromotion.kt:54]: ditto

**Review by joshliebe — APPROVED** [review]
Minor comments, works as expected and code looks good ✅

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/promotion/passwords/SyncPasswordsPromotion.kt (hunk 1 above)
Comment (CDRussell) [reply sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/promotion/passwords/SyncPasswordsPromotion.kt]: Good idea, and i see now that's the pattern for other priority key definitions. Done!

**Pull request merged**